Firestore/core/test/unit/core/expressions/comparison_test.cc

```diff
@@ -14,60 +14,921 @@
+                Returns(testutil::Value(false)))
+        << "lt(" << ExprId(pair.first) << ", " << ExprId(pair.second) << ")";
+  }
+}
+
+// --- Specific Lt Tests ---
+
+TEST_F(LtFunctionTest, NullOperandReturnsNull) {
+  for (const auto& val :
+       ComparisonValueTestData::AllSupportedComparableValues()) {
+    EXPECT_THAT(
+        EvaluateExpr(*LtExpr({SharedConstant(model::NullValue()), val})),
+        ReturnsNull())
+        << "lt(null, " << ExprId(val) << ")";
+    EXPECT_THAT(
+        EvaluateExpr(*LtExpr({val, SharedConstant(model::NullValue())})),
+        ReturnsNull())
+        << "lt(" << ExprId(val) << ", null)";
+  }
+  EXPECT_THAT(EvaluateExpr(*LtExpr({SharedConstant(model::NullValue()),
+                                    SharedConstant(model::NullValue())})),
+              ReturnsNull());
+  EXPECT_THAT(
+      EvaluateExpr(*LtExpr({SharedConstant(model::NullValue()),
+                            std::make_shared<api::Field>("nonexistent")})),
+      ReturnsUnset());
+}
 
-TEST(Eq, Basic) {
-  auto result = eq(api::Field("foo"), constant(42))
-                    .ToEvaluable()
-                    ->Evaluate(NewContext(), Doc("docs/1", 0, Map("foo", 42)));
+TEST_F(LtFunctionTest, NaNComparisonsReturnFalse) {
+  auto nan_expr = SharedConstant(std::numeric_limits<double>::quiet_NaN());
+  EXPECT_THAT(EvaluateExpr(*LtExpr({nan_expr, nan_expr})),
+              Returns(testutil::Value(false)));
+
+  for (const auto& num_val : ComparisonValueTestData::NumericValues()) {
+    EXPECT_THAT(EvaluateExpr(*LtExpr({nan_expr, num_val})),
+                Returns(testutil::Value(false)))
+        << "lt(NaN, " << ExprId(num_val) << ")";
+    EXPECT_THAT(EvaluateExpr(*LtExpr({num_val, nan_expr})),
+                Returns(testutil::Value(false)))
+        << "lt(" << ExprId(num_val) << ", NaN)";
+  }
+  for (const auto& other_val :
+       ComparisonValueTestData::AllSupportedComparableValues()) {
+    bool is_numeric = false;
+    for (const auto& num_val : ComparisonValueTestData::NumericValues()) {
+      if (other_val == num_val) {
+        is_numeric = true;
+        break;
+      }
+    }
+    if (!is_numeric) {
+      EXPECT_THAT(EvaluateExpr(*LtExpr({nan_expr, other_val})),
+                  Returns(testutil::Value(false)))
+          << "lt(NaN, " << ExprId(other_val) << ")";
+      EXPECT_THAT(EvaluateExpr(*LtExpr({other_val, nan_expr})),
+                  Returns(testutil::Value(false)))
+          << "lt(" << ExprId(other_val) << ", NaN)";
+    }
+  }
+  EXPECT_THAT(
+      EvaluateExpr(*LtExpr({SharedConstant(testutil::Array(testutil::Value(
+                                std::numeric_limits<double>::quiet_NaN()))),
+                            SharedConstant(testutil::Array(testutil::Value(
+                                std::numeric_limits<double>::quiet_NaN())))})),
+      Returns(testutil::Value(false)));
+}
+
+TEST_F(LtFunctionTest, ErrorHandling) {
```

Comment:
These ErrorHandling tests seem wrong to me.

Reading the backend code, Unset is only returned from Field. After that, any operator that tries to use an Unset returns an Error. So Unsets very quickly becomes errors.